modules/pam_timestamp/hmac_openssl_wrapper.c

```diff
@@ -184,7 +190,7 @@ write_file(pam_handle_t *pamh, const char *file_name, char *text,
     bytes_written = pam_modutil_write(fd, text, text_length);
     close(fd);
 
-    if (bytes_written < text_length) {
+    if (bytes_written < 0 || (size_t)bytes_written < text_length) {
         pam_syslog(pamh, LOG_ERR, "Short write on %s", file_name);
         free(text);
```

Comment:
Inconsistent handling of `text` buffer. All other locations have `pam_overwrite_n(text, text_length);`, thus not having it here looks strange. That aside: Overwriting an input buffer on error looks strange AND is at least "unexpected behaviour". Overwriting buffer contents should be tasked to the caller, as changing a buffer in a write function is at least confusing.